File: #1007.minimum-domino-rotations-for-equal-row.cpp
class Solution {
public:
    int minDominoRotations(vector<int>& tops, vector<int>& bottoms) {

        int size = tops.size();
        int top = 0;
        int bottom = 0;
        for (int i = 0; i < size; i++) {
            if (tops[i] == tops[0] || bottoms[i] == tops[0]) {
                if (tops[i] != tops[0]) {
                    top++;
                }
                if (bottoms[i] != tops[0]) {
                    bottom++;
                }
            } else {
                break;
            }

            if (i == size-1) {
                return std::min(top, bottom);
            }
        }

        top = 0;
        bottom = 0;
        for (int i = 0; i < size; i++) {
            if (tops[i] == bottoms[0] || bottoms[i] == bottoms[0]) {
                if (tops[i] != bottoms[0]) {
                    top++;
                }
                if (bottoms[i] != bottoms[0]) {
                    bottom++;
                }
            } else {
                break;
            }

            if (i == size - 1) {
                return std::min(top, bottom);
            }
        }
        return -1;
    }
};

File: #1007.minimum-domino-rotations-for-equal-row.js
/**
 * @param {number[]} tops
 * @param {number[]} bottoms
 * @return {number}
 */
var minDominoRotations = function(tops, bottoms) {
    const numberOfA = new Array(7).fill(0);
    const numberOfB = new Array(7).fill(0);
    const same = new Array(7).fill(0);
    for (let i = 0; i < tops.length; i++) {
        numberOfA[tops[i]]++;
        numberOfB[bottoms[i]]++;
        if (tops[i] == bottoms[i]) {
            same[tops[i]]++;
        }
    }
    for (let i = 1; i <= 6; i++) {
        if (numberOfA[i] + numberOfB[i] - same[i] == tops.length) {
            return Math.min(numberOfA[i] - same[i],
                numberOfB[i] - same[i]);
        }
    }
    return -1;
};
